src/ec/packages/ec-component-checkbox/checkbox-group.html.twig

```diff
@@ -0,0 +1,102 @@
+{% spaceless %}
+
+{#
+  Parameters:
+  - "label_id" (string) (default: '')
+  - "items" (array) (default: []): array of checkbox_button
+  - "name" (string) (default: '')
+  - "helper_id" (string) (default: '')
+  - "hide_label" (boolean) (default: false)
+  - "invalid" (boolean) (default: false)
+  - "required" (boolean) (default: false)
+  - "extra_classes" (optional) (string) (default: '')
+  - "extra_attributes" (optional) (array) (default: [])
+    - "name" (string) Attribute name, eg. 'data-test'
+    - "value" (string) Attribute value, eg: 'data-test-1'
+  Blocks:
+  - "helper_text"
+  - "invalid_text"
+  - "optional_text"
+  - "required_text"
+  - "label"
+#}
+
+{# Internal properties #}
+
+{% set _label_id = label_id|default('checkbox-default-label') %}
+{% set _items = items|default([]) %}
+{% set _name = name|default('') %}
+{% set _helper_id = helper_id|default('') %}
+{% set _hide_label = hide_label|default(false) %}
+{% set _invalid = invalid|default(false) %}
+{% set _required = required|default(false) %}
+
+{% set _helper_text = helper_text|default('') %}
+{% set _invalid_text = invalid_text|default('') %}
+{% set _required_text = required_text|default('*') %}
+{% set _optional_text = optional_text|default('') %}
+{% set _label = label|default('') %}
+
+{% set _css_class = 'ecl-form-group ecl-form-group--checkbox' %}
+{% set _extra_attributes = '' %}
+
+{# Internal logic - Process properties #}
+
+{% if extra_classes is defined and extra_classes is not empty %}
+  {% set _css_class = _css_class ~ ' ' ~ extra_classes %}
+{% endif %}
+
+{% if extra_attributes is defined and extra_attributes is not empty and extra_attributes is iterable %}
+  {% for attr in extra_attributes %}
+    {% set _extra_attributes = _extra_attributes ~ ' ' ~ attr.name ~ '="' ~ attr.value ~ '"' %}
+  {% endfor %}
+{% endif %}
+
+{# Print the result #}
+
+<fieldset
+  class="{{ _css_class }}"
+  aria-labelledby="{{ _label_id }}"
+  {{ _helper_id is not empty ? "aria-describedby=\"" ~ _helper_id ~ "\""}}
+  {{ _extra_attributes|raw }}
+>
+
+  {%- if _label is not empty %}
+    <legend
+      class="ecl-form-label{{ hide_label ? ' ecl-form-label--hidden'}}{{ _invalid_text ? ' ecl-checkbox__group--invalid'}}"
```

Comment:
@planctus As I started form the actual Radio component, I keep this option for checkboxes. Maybe I have to validate with the platform team if it's still necessary. what do you think?

Reply:
as we agreed together, we don't introduce properties not being present in the ECL components here, but i share with you some concerns regarding the definition of these "groups" of form fields, if we render a fieldset we might have (or it would be better...) to have also a label basing on this:
https://fae.disability.illinois.edu/rulesets/CONTROL_8/
So the logic might change in the future, but let's follow ECL so that the components will stay aligned, if we need to do something on them we will do that in all the "places".

Reply:
@planctus Ok thank you for clarification, I have removed the hide_label parameter for now.